File: MapGUI3/src/Model/Output.java
package Model;

import java.util.ArrayList;
import java.util.List;

public class Output<T> implements IList<T> {
    private List<T> list;

    //constructor for Output class
    public Output(){list=new ArrayList<>();}

    //function that adds an element in the list
    public void add(T element) {list.add(element);}

    public T pop() {T v=list.get(0); list.remove(v); return v;}

    //function that eliminates all the elements from the list
    public void clear() {list.clear();}

    public boolean isEmpty() { return list.isEmpty();}

    public String toString(){return "Output = " + list.toString();}
}

File: MapA7/MapA7/Model/Output.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MapA7.Model
{
    public class Output:IList
    {
        private List<String> output;

        public Output() { this.output = new List<String>(); }

        public void Add(String element) { this.output.Add(element); }

        public void Clear() { this.output.Clear(); }

        public String Tostring()
        {
            String res = "[";
            foreach (String i in output)
            {
                res = res + i + ',';
            }
            if(res.Length>2) res=res.Remove(res.Length - 1);
            res = res + ']';
            return "Output = " + res;
        }
    }
}
